### PR TITLE
Resolves some more build warnings on Library project

### PR DESCRIPTION
**Pull request by valadas**
This PR focuses a bit on Lists and some related upstream classes/interfaces.

Any feedback is welcome

### CODE REVIEW

File: DNN Platform/Library/Common/Internal/IGlobals.cs
Comment (kieran23101) [inline DNN Platform/Library/Common/Internal/IGlobals.cs:34]: Small spelling mistake - "subforlder" to "subfolder"

Other than that, looks good. Nice job

Comment (valadas) [reply DNN Platform/Library/Common/Internal/IGlobals.cs]: Thank you, corrected in https://github.com/dnnsoftware/Dnn.Platform/pull/4243/commits/05d426fdfeb31929b2333fa20ebab36f4ea563f0

File: DNN Platform/Library/Common/Lists/CachedCountryList.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/CachedCountryList.cs:79]:
```suggestion
        /// <summary>Represents a country.</summary>
```

File: DNN Platform/Library/Common/Internal/TestableGlobals.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Internal/TestableGlobals.cs:13]:
```suggestion
    /// <summary>A service locator for <see cref="IGlobals"/>.</summary>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:42]: Per [CA1819](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1819), properties shouldn't return arrays (because they would allow editing the underlying data).
```suggestion
        public IReadOnlyCollection<string> UnLocalizedLists => UnLocalizableLists;
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:80]: Did we lose the `!` here?
```suggestion
            if (Thread.CurrentThread.CurrentCulture.Name != Localization.SystemLocale && !UnLocalizableLists.Contains(listEntry.ListName))
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:113]:
```suggestion
        /// Deletes a list.
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:115]:
```suggestion
        /// <param name="listName">The name of the list to delete.</param>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:188]:
```suggestion
        /// <param name="deleteChild">A value indicating whether to also delete the children of that item.</param>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:202]:
```suggestion
        /// <summary>Deletes a list entry by its name.</summary>
        /// <param name="listName">The name of the list entry.</param>
        /// <param name="listValue">The value of the list entry.</param>
        /// <param name="deleteChild">A value indicating whether to also delete the children of that item.</param>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:245]:
```suggestion
        /// <summary>Gets the entries in the list with the given <paramref name="listName"/>.</summary>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:255]:
```suggestion
        /// <summary>Gets the entries in a child list.</summary>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:266]:
```suggestion
        /// <summary>Gets the entries in a child list.</summary>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:280]:
```suggestion
        /// <returns>A dictionary where the index is <see cref="ListEntryInfo.Key"/> and the value is the actual <see cref="ListEntryInfo"/>.</returns>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:291]:
```suggestion
        /// <returns>A dictionary where the index is <see cref="ListEntryInfo.Key"/> and the value is the actual <see cref="ListEntryInfo"/>.</returns>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:303]:
```suggestion
        /// <returns>A dictionary where the index is <see cref="ListEntryInfo.Key"/> and the value is the actual <see cref="ListEntryInfo"/>.</returns>
```

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:467]:
```suggestion
        /// <param name="moveUp">The entry is moved up if <c>true</c> or moved down if <c>false</c>.</param>
```

File: DNN Platform/Library/Common/Lists/ListEntryInfoCollection.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListEntryInfoCollection.cs:43]:
```suggestion
        /// <summary>Gets the entry by its key.</summary>
```

File: DNN Platform/Library/Common/Lists/ListEntryInfoCollection.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListEntryInfoCollection.cs:70]:
```suggestion
        /// <summary>Gets the children by its parent name.</summary>
```

File: DNN Platform/Library/Common/Lists/ListInfo.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListInfo.cs:13]:
```suggestion
    /// <summary>Represents the information relative to a list.</summary>
```

File: DNN Platform/Library/Common/Lists/ListInfo.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListInfo.cs:44]:
```suggestion
        /// <summary>Gets the display name, which includes the <see cref="Parent"/> list and this list's <see cref="Name"/>.</summary>
```

File: DNN Platform/Library/Common/Lists/ListInfo.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListInfo.cs:61]:
```suggestion
        /// <summary>Gets a unique key to identify this list, which includes the <see cref="ParentKey"/> and this list's <see cref="Name"/>.</summary>
```

File: DNN Platform/Library/Common/Lists/ListInfo.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListInfo.cs:93]:
```suggestion
        /// <summary>Gets or sets a value indicating whether this instance is populated.</summary>
```

**Review by bdukes — APPROVED** [review]
Thanks!

**Comment by valadas** [comment]
I will revisit this after #4244 is approved and merged so adjust it's usage here too and will look at the rest of @bdukes suggestions shortly. I put it on hold to prevent an accidental merge.

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (valadas) [reply DNN Platform/Library/Common/Lists/ListController.cs:80]: You are correct

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (valadas) [reply DNN Platform/Library/Common/Lists/ListController.cs:243]: Oh, just learned about paramref, nice...

**Comment by valadas** [comment]
@bdukes as discussed in #4242 I have refactored this to use DI for eventLogger in https://github.com/dnnsoftware/Dnn.Platform/pull/4243/commits/639ae56c6eaf0c9879e25c64d6b4f175766fe4bd

Also merged all comments corrections above, I think it's good to re-review now. Thanks for all the feedback

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (bdukes) [inline DNN Platform/Library/Common/Lists/ListController.cs:49]:
```suggestion
        /// <summary>Initializes a new instance of the <see cref="ListController"/> class.</summary>
        public ListController()
            : this(null)
        {
        }

        /// <summary>Initializes a new instance of the <see cref="ListController"/> class.</summary>
        /// <param name="eventLogger">An event logger.</param>
        public ListController(IEventLogger eventLogger)
        {
            this.eventLogger = eventLogger ?? Globals.DependencyProvider.GetRequiredService<IEventLogger>();
        }

```

**Review by bdukes — APPROVED** [review]

File: DNN Platform/Library/Common/Lists/ListController.cs
Comment (valadas) [reply DNN Platform/Library/Common/Lists/ListController.cs:49]: Hmm, I like that :)

**Comment by valadas** [comment]
I also have 2 tests to adjust for DI, I'll handle that shortly...

**Comment by valadas** [comment]
I rebased on latest changes from develop and the tests should now be fixed (once it builds)

**Review by david-poindexter — APPROVED** [review]
Looks great @valadas - thank you! 🎉

**Pull request merged**
